Sources/Playbook/Components/Typeahead/TypeaheadCatalog.swift

```diff
@@ -14,31 +14,39 @@ public struct TypeaheadCatalog: View {
     @State private var assetsUsers = Mocks.multipleUsersDictionary
     @State private var searchTextUsers: String = ""
     @State private var searchTextColors: String = ""
+    @State private var searchText: String = ""
     @State private var searchTextDebounce: String = ""
-    @State private var searchTextDebounce2: String = ""
     @State private var didTapOutside: Bool? = false
-    var popoverManager = PopoverManager()
+    @State private var isPresented1: Bool = false
+    @State private var presentDialog: Bool = false
+    @State private var isLoading: Bool = false
+    @State private var assetsUser = Mocks.multipleUsersDictionary
     @FocusState var isFocused1
     @FocusState var isFocused2
+
+    var popoverManager = PopoverManager()
     
     public var body: some View {
         PBDocStack(title: "Typeahead") {
             PBDoc(title: "Default", spacing: Spacing.small) { colors }
             PBDoc(title: "With Pills", spacing: Spacing.small) { users }
-            //      PBDoc(title: "Debounce", spacing: Spacing.small) { debounce }
+            #if os(macOS)
+            PBDoc(title: "Dialog") { dialog }
+            #endif
         }
         .onTapGesture {
             isFocused1 = false
             isFocused2 = false
         }
-        .popoverHandler()
+        .popoverHandler(id: 1)
+        .popoverHandler(id: 2)
```

Comment:
Do we need two different handlers here ?

Reply:
yes. We need to manage each popover separetly so we can present different popovers in different view hierarchies.